tutorials/misc/plot_ecog.py

```diff
@@ -183,20 +166,118 @@ def animate(i, activity):
+# This time we will also pass the plot title so that can be updated
+# with information from the annotations.
+def init():
+    """Create an empty frame."""
+    return paths, title
+
+
+def animate(i, activity, events):
+    """Animate the plot."""
+    paths.set_array(activity[:, i])
+    tsl2 = slice(i + start_sample - sfreq, i + start_sample + sfreq)
+    for j, tpath in enumerate(tpaths):
+        tpath.set_data(raw_notched.times[tsl], ts_data[j, tsl2])
+    # If this sample contains an annotation (for a seizure-related behavior)
+    # then change the title of the plot
+    if i + start_sample in events[:, 0]:
+        # Currently this doesn't replace the text, but writes over it.
+        # This needs fixing
+        event_idx = np.argwhere(events[:, 0] == (i + start_sample))[0][0]
+        title.set_text(inv_event_id[events[event_idx, 2]])
+        fig.canvas.draw()  # force redrawing
+    return paths, title
+
+
+# create the figure and apply the animation of the
+# raw time series data
+fig, ax = plt.subplots(figsize=(5, 5))
+tax = fig.add_axes([0.15, 0.02, 0.6, 0.15])
+ax.imshow(im)
+ax.set_axis_off()
+tax.set_axis_off()
+
+# We want the mid-point (0 uV) to be white, so we will scale from -vmax to vmax
+# so that negative voltages are blue and positive voltages are red
+ts_data = raw_notched.get_data()
+vmax = np.percentile(ts_data, 90)
+paths = ax.scatter(*xy_pts.T, c=np.zeros(len(xy_pts)), s=40,
+                   cmap=cmap, vmin=-vmax, vmax=vmax)
+ax.set_xlim([0, im.shape[0]])
+ax.set_ylim([im.shape[1], 0])
+tsl = slice(start_sample - sfreq, start_sample + sfreq)
+tpaths = tax.plot(raw_notched.times[tsl], ts_data[:, tsl].T)
+fig.colorbar(paths, ax=ax)
+title = ax.set_title('iEEG voltage over time', size='large')
+
+# this will be a much longer animation because the seizure is quite long
+sl = slice(start_sample, ts_data.shape[1])
+show_power = ts_data[:, sl]
+anim = animation.FuncAnimation(fig, animate, init_func=init,
+                               fargs=(show_power, events),
+                               frames=show_power.shape[1],
+                               interval=20, blit=True)
+
+
 ###############################################################################
 # Alternatively, we can project the sensor data to the nearest locations on
 # the pial surface and visualize that:
 
-# sphinx_gallery_thumbnail_number = 4
+# sphinx_gallery_thumbnail_number = 5
+
+# decimate by a factor of 10 for speed here
+ts_data = raw.get_data()
+sl = slice(start_sample, ts_data.shape[1], 10)
 
 evoked = mne.EvokedArray(
     gamma_power_t[:, sl], raw.info, tmin=raw.times[sl][0])
-stc = mne.stc_near_sensors(evoked, trans, subject, subjects_dir=subjects_dir)
+src = mne.read_source_spaces(
+    op.join(subjects_dir, 'fsaverage', 'bem', 'fsaverage-ico-5-src.fif'))
+trans = None  # identity transform
+stc = mne.stc_near_sensors(evoked, trans, 'fsaverage', src=src,
+                           subjects_dir=subjects_dir)
```

Comment:
running locally now I see on this line:
```
  File "<decorator-gen-283>", line 24, in stc_near_sensors
  File "/opt/mne/python/mne/source_estimate.py", line 3301, in stc_near_sensors
    evoked.drop_channels(nan_chs)
  File "/opt/mne/python/mne/channels/channels.py", line 920, in drop_channels
    return self._pick_drop_channels(idx)
  File "/opt/mne/python/mne/channels/channels.py", line 943, in _pick_drop_channels
    pick_info(self.info, idx, copy=False)
  File "<decorator-gen-8>", line 24, in pick_info
  File "/opt/mne/python/mne/io/pick.py", line 539, in pick_info
    raise ValueError('No channels match the selection.')
ValueError: No channels match the selection.
```

Reply:
I don't actually get that error... I somehow get to this, but I just assumed that was cuz my Pycharm IDE is glitchy showing all the figures:

```
Traceback (most recent call last):
  File "/Users/adam2392/Documents/mne-python/tutorials/misc/plot_ecog.py", line 279, in <module>
    brain.show_view(azimuth=-20, elevation=60)
TypeError: show_view() got an unexpected keyword argument 'azimuth'
```

Reply:
Hmm, this worked for me last time, but double checking now

Reply:
Fixed it